src/runtime_src/core/tools/xbflash2/OO_Program_Qspips.cpp

```
@@ -0,0 +1,295 @@
+
+    if (!force && !XBU::can_proceed())
+        throw std::errc::operation_canceled;
+
+    pcidev::pci_device dev(bdf, bar, baroff, flash_type);
+    XQSPIPS_Flasher qspips(&dev);
+
+    return qspips.xclUpgradeFirmware(bin, offset);
+}
+
+int 
+qspips_erase(po::variables_map& vm) {
+    std::string bdf;
+    std::string flash_type;
+    std::string soffset;
+    std::string slength;
+    std::string output;
+    size_t offset = 0, len = GOLDEN_BASE;
+    bool force = false;
+    std::string sBar, sBarOffset;
+    size_t baroff = INVALID_OFFSET;
+    int bar = 0;    
+
+    XBU::sudo_or_throw_err();
+    //mandatory command options
+    try {
+        bdf = vm["device"].as<std::string>();
+    }
+    catch (...) {
+        throw std::errc::invalid_argument;
+    }
+
+    //optionals command options
+    try {
+        if (vm.count("flash-part"))
+            flash_type = vm["flash-part"].as<std::string>();
+        if (vm.count("offset")) {
+            soffset = vm["offset"].as<std::string>();
+            if (!soffset.empty()) {
+                std::stringstream sstream(soffset);
+                if (offset != INVALID_OFFSET)
+                    sstream >> offset;
+            }
+        }
+        if (vm.count("length")) {
+            slength = vm["length"].as<std::string>();
+            if (!slength.empty()) {
+                std::stringstream sstream(slength);
+                sstream >> len;
+            }
+        }
+    }
+    catch (...) {
+    }
+
+    //optionals command line args
+    try {
+        if (vm.count("bar")) {
+            sBar = vm["bar"].as<std::string>();
+            if (!sBar.empty())
+                bar = std::stoi(sBar);
+        }
+
+        if (vm.count("bar-offset")) {
+            sBarOffset = vm["bar-offset"].as<std::string>();
+            if (!sBarOffset.empty()) {
+                std::stringstream sstream(sBarOffset);
+                sstream >> baroff;
+            }
+        }
+    }
+    catch (...) {
```

Please don't use empty catch blocks.